src/tests/specialfunctions/test_specialfunctions_gamma.fypp

```diff
@@ -0,0 +1,587 @@
+
+            call check(error, gamma(x(i)), ans(i), "Complex kind ${k1}$ failed",&
+                       thr = tol_${k1}$, rel = .true.)
+
+        end do
+
+        #:endif
+    end subroutine test_gamma_${t1[0]}$${k1}$
+
+
+
+    subroutine test_loggamma_${t1[0]}$${k1}$(error)
+        type(error_type), allocatable, intent(out) :: error
+        integer, parameter :: n = 4
+        integer :: i
+
+        #:if k1 == "int8"
+
+        ${t1}$, parameter :: x(n) = [1_${k1}$, 2_${k1}$, 10_${k1}$, 47_${k1}$]
+        real(sp), parameter :: ans(n) = [0.0, 0.0, 1.28018274e1, 1.32952575e2]
+
+        #:elif k1 == "int16"
+
+        ${t1}$, parameter :: x(n) = [1_${k1}$, 2_${k1}$, 111_${k1}$, 541_${k1}$]
+        real(sp), parameter :: ans(n) = [0.0, 0.0, 4.10322777e2, 2.86151221e3]
+
+        #:elif k1 == "int32"
+
+        ${t1}$, parameter :: x(n) = [1_${k1}$, 2_${k1}$, 2021_${k1}$,          &
+                                     42031_${k1}$]
+        real(sp), parameter :: ans(n) = [0.0, 0.0, 1.33586470e4, 4.05433461e5]
+
+        #:elif k1 == "int64"
+
+        ${t1}$, parameter :: x(n) = [1_${k1}$, 2_${k1}$, 2021_${k1}$,          &
+                                     42031_${k1}$]
+        real(sp), parameter :: ans(n) = [0.0, 0.0, 1.33586470e4, 4.05433461e5]
+
+        #:elif t1[0] == "c"
+
+        ${t1}$, parameter :: x(n) = [(0.25_${k1}$, 0.25_${k1}$),               &
+                                     (0.5_${k1}$, -0.5_${k1}$),                &
+                                     (1.0_${k1}$, 1.0_${k1}$),                 &
+                                     (-1.254e1_${k1}$, -9.87_${k1}$)]
+
+        ${t1}$, parameter :: ans(n) =                                          &
+                    [(0.90447450949333889_${k1}$, -0.83887024394321282_${k1}$),&
+                     (0.11238724280962311_${k1}$, 0.75072920212205074_${k1}$), &
+                    (-0.65092319930185634_${k1}$, -0.30164032046753320_${k1}$),&
+                   (-4.7091788015763380e1_${k1}$, 1.4804627819235690e1_${k1}$)]
+        #:endif
+
+
+        #:if t1[0] == "i"
+
+        do i = 1, n
+
+            call check(error, log_gamma(x(i)), ans(i), "Integer kind ${k1}$ "  &
+              //"failed", thr = tol_sp, rel = .true.)
+
+        end do
+
+        #:elif t1[0] == "c"
+
+        do i = 1, n
+
+            call check(error, log_gamma(x(i)), ans(i), "Complex kind ${k1}$ "  &
+              //"failed", thr = tol_${k1}$, rel = .true.)
+
+        end do
+
+        #:endif
+    end subroutine test_loggamma_${t1[0]}$${k1}$
+
+    #:endfor
+
+
+
+
+    #:for k1, t1 in INT_KINDS_TYPES
+      #:for k2, t2 in REAL_KINDS_TYPES
+
+    subroutine test_lincgamma_${t1[0]}$${k1}$${k2}$(error)
+        type(error_type), allocatable, intent(out) :: error
+        integer, parameter :: n = 4
+        integer :: i
+        ${t1}$ :: p(n) = [1_${k1}$, 2_${k1}$, 3_${k1}$, 2_${k1}$]
+        ${t2}$ :: x(n) = [0.5_${k2}$, 3.5_${k2}$, -5.0_${k2}$, -10.0_${k2}$]
```

Comment:
```suggestion
        ${t1}$, parameter :: p(n) = [1_${k1}$, 2_${k1}$, 3_${k1}$, 2_${k1}$]
        ${t2}$, parameter :: x(n) = [0.5_${k2}$, 3.5_${k2}$, -5.0_${k2}$, -10.0_${k2}$]
```

Reply:
If not parameter, then it should be assigned outside the declaration statements.